src/fides/api/util/connection_type.py

```diff
@@ -181,6 +181,11 @@ def get_connection_type_secret_schema(*, connection_type: str) -> dict[str, Any]
     }
     if schema.get("additionalProperties"):
         schema.pop("additionalProperties")
+
+    # set an empty array for the 'required' key, if there are no required fields on the schema
+    # this helps the FE, which is expecting _some_ value here, even if it's an empty list
+    if "required" not in schema:
+        schema["required"] = []
```

Comment:
as the comment says, this is basically a fallback, since having _no_ `required` field on our response seems to cause problems for the FE...

Reply:
when did you run into this?

Reply:
ah it's the other ticket! https://ethyca.atlassian.net/browse/PROD-2722 I'm caught up

Reply:
This is that not-so-great endpoint where we were returning the pydantic v1 schema and now there's some changes to try to backport the pydantic v2 schema to return the pydantic v1 version so the UI doesn't have to change. Ideally we'd rework the backend and frontend together at some point.

Reply:
makes sense, agreed that it'd be nice to refactor!